main.cc

```diff
@@ -1032,6 +1034,22 @@ int main(int ac, char** av) {
                 repair_uninit_messaging_service_handler().get();
             });
 
+            supervisor::notify("starting CDC Generation Management service");
+            /* This service uses the system distributed keyspace.
+             * It will only do that *after* it has joined the token ring, and the token ring joining
+             * procedure (`storage_service::init_server`) is responsible for initializing sys_dist_ks.
+             *
+             * However, sys_dist_ks is stopped *before* CGM is stopped (`storage_service::drain_on_shutdown` below),
+             * so CGM takes sharded<db::sys_dist_ks> and must check local_is_initialized() every time it accesses it,
```

Comment:
I understand why local_shared() has to be called to prevent destruction of sys_dist_ks but why do we have to check local_is_initialized()? Won't `storage_service::init_server` initialise sys_dist_ks before calling CGM?

Reply:
as the comment explains, the issue is not that of starting the services, it is that of shutting them down.
`sys_dist_ks` is stopped before generation management service, so the latter must check `local_is_initialized` because `sys_dist_ks` may be stopped already when it tries to access it

Reply:
The comment wasn't clear enough for me to be 100% sure that it talks only about the stopping. `local_is_initialized` makes an impression that starting is involved as well. Please rephrase to leave no doubt what exactly has to be checked and when.